src/Symfony/Component/HttpFoundation/InputBag.php

```diff
@@ -17,14 +17,20 @@
 /**
  * InputBag is a container for user input values such as $_GET, $_POST, $_REQUEST, and $_COOKIE.
  *
+ * @template TInput of string|int|float|bool|null
+ *
  * @author Saif Eddin Gmati <azjezz@protonmail.com>
  */
 final class InputBag extends ParameterBag
 {
     /**
      * Returns a scalar input value by name.
      *
-     * @param string|int|float|bool|null $default The default value if the input key does not exist
+     * @template TDefault of TInput|null
```

Comment:
this is a bit restrictive, we could also say  `@template TDefault of string|int|float|bool|null`
this could allow one to have an `InputBag<string|null>` and call `$bag->get('foo', 123)` to know when the default is used, aka the value is not found.
Nitpicking :)

Reply:
@azjezz let's update this?